releases/release-4.0.16.md

```diff
@@ -0,0 +1,112 @@
+        - 将 Kafka Sink `partition-num` 的默认值改为 3，使 TiCDC 更加平均地分发消息到各个 Kafka partition [#3337](https://github.com/pingcap/ticdc/issues/3337)
+
+## 提升改进
+
++ TiDB
+
+    - 升级 Grafana 到 v7.5.11，规避老版本的安全漏洞
+
++ TiKV
+
+    - 当使用 Backup & Restore 恢复数据或使用 TiDB Lightning 的 Local-backend 导入数据时，采用 zstd 算法压缩 SST 文件，从而减小磁盘使用空间 [#11469](https://github.com/tikv/tikv/issues/11469)
+
++ Tools
+
+    + Backup & Restore (BR)
+
+        - 增强恢复的鲁棒性 [#27421](https://github.com/pingcap/tidb/issues/27421)
+
+    + TiCDC
+
+        - 为 EtcdWorker 添加 tick 频率限制，防止 PD 的 etcd 写入次数过于频繁影响 PD 服务 [#3112](https://github.com/pingcap/ticdc/issues/3112)
+        - 优化 TiKV 重新加载时的速率限制控制，缓解 changefeed 初始化时 gPRC 的拥堵问题 [#3110](https://github.com/pingcap/ticdc/issues/3110)
+
+## Bug 修复
+
++ TiDB
+
+    - 修复在统计信息模块的估算代价中，当执行 range 转 points 时由于数值溢出而导致的查询崩溃 [#23625](https://github.com/pingcap/tidb/issues/23625)
+    - 修复当 `ENUM` 类型作为 `IF` 或 `CASE WHEN` 等控制函数的参数时，返回结果不正确的问题 [#23114](https://github.com/pingcap/tidb/issues/23114)
+    - 修复由于 `tidb_enable_vectorized_expression` 设置的值不同（`on` 或 `off`）导致 `GREATEST` 函数返回结果不一致的问题 [#29434](https://github.com/pingcap/tidb/issues/29434)
+    - 修复 `Index Join` 在使用前缀索引时某些情况下崩溃的问题 [#24547](https://github.com/pingcap/tidb/issues/24547)
+    - 修复在某些情况下 Planner 可能缓存无效 `join` 计划的问题 [#28087](https://github.com/pingcap/tidb/issues/28087)
+    - 修复当 `sql_mode` 为空时，TiDB 无法插入 `null` 到非 null 列的问题 [#11648](https://github.com/pingcap/tidb/issues/11648)
+    - 修正 `GREATEST` 和 `LEAST` 函数的返回值类型错误 [#29019](https://github.com/pingcap/tidb/issues/29019)
+    - 修复 `grant` 和 `revoke` 操作在授予和撤销全局权限时，报 `privilege check fail` 错误的问题 [#29675](https://github.com/pingcap/tidb/issues/29675)
+    - 修复当 `CASE WHEN` 函数和 `ENUM` 类型一起使用时的崩溃问题 [#29357](https://github.com/pingcap/tidb/issues/29357)
+    - 修复 `microsecond` 函数的向量化表达式版本结果不正确的问题 [#29244](https://github.com/pingcap/tidb/issues/29244)
+    - 修复 `hour` 函数在向量化表达式中执行结果错误的问题 [#28643](https://github.com/pingcap/tidb/issues/28643)
+    - 修复乐观事务冲突可能导致事务相互阻塞的问题 [#11148](https://github.com/tikv/tikv/issues/11148)
+    - 修复 `auto analyze` 输出的日志信息不完整的问题 [#29188](https://github.com/pingcap/tidb/issues/29188)
+    - 修复当 `SQL_MODE` 为 'NO_ZERO_IN_DATE' 时，使用非法的默认时间不报错的问题 [#26766](https://github.com/pingcap/tidb/issues/26766)
+    - 修复 Grafana 上 Coprocessor Cache 监控面板不显示数据的问题。现在 Grafana 会显示 `hits`/`miss`/`evict` 的数据 [#26338](https://github.com/pingcap/tidb/issues/26338)
+    - 修复并发 truncate 同一个分区会导致 DDL 语句执行卡住的问题 [#26229](https://github.com/pingcap/tidb/issues/26229)
+    - 修复 `CONCAT` 函数的参数为负浮点数时，该参数在返回结果中被截断末位数的问题 [#29417](https://github.com/pingcap/tidb/issues/29417)
+    - 修复使用 `NATUAL JOIN` 连接多张表时，查询结果中多出一列的问题 [#29481](https://github.com/pingcap/tidb/issues/29481)
+    - 修复 `IndexScan` 使用前缀索引时，`TopN` 被错误下推至 `indexPlan` 的问题 [#29711](https://github.com/pingcap/tidb/issues/29711)
+    - 修复在 `DOUBLE` 类型的自增列上重试事务会导致数据错误的问题 [#29892](https://github.com/pingcap/tidb/issues/29892)
```

Comment:
- 修复 `Decimal` 转为 `String` 时长度信息计算错误的问题 [#29417](https://github.com/pingcap/tidb/issues/29417)

Reply:
A new added one?

Reply:
Looks like this description is not so accurate. Because other expr's behavior can also be affected, check [here](https://github.com/pingcap/tidb/issues/29417#issuecomment-977721045)